musicbot/bot.py

```diff
@@ -1223,30 +1223,49 @@ def _gen_embed(self):
         If a command is specified, it prints a help message for that command.
         Otherwise, it lists the available commands.
         """
+        self.commands = []
+        prefix = self.config.command_prefix
 
-        if command:
-            cmd = getattr(self, 'cmd_' + command, None)
-            if cmd and not hasattr(cmd, 'dev_cmd'):
-                return Response(
-                    "```\n{}```".format(
-                        dedent(cmd.__doc__)
-                    ).format(command_prefix=self.config.command_prefix),
-                    delete_after=60
-                )
-            else:
-                raise exceptions.CommandError(self.str.get('cmd-help-invalid', "No such command"), expire_in=10)
-
-        else:
-            commands = []
+        async def gen_cmd_list(message, all=False):
 
             for att in dir(self):
                 if att.startswith('cmd_') and att != 'cmd_help' and not hasattr(getattr(self, att), 'dev_cmd'):
```

Comment:
There's probably no reason to actually exclude the help command here, even though they're using it to check these commands. Including the help command ensures granularity.